### PR TITLE
mgr/cephadm: fixing call method timeout handling

### PR DESCRIPTION
**Pull request by rkachach**
The root cause was `asyncio.gather` being called before `asyncio.wait_for` so as consequence in case the created process hangs, cephadm will hangs also forever (calling the `asyncio.gather`). The fix consists on switching the order of these calls.

Note: as part of this fix I'm reducing the maximum tries for mon from 15 (default) to 3. So at most cephadm will wait 3 min for the mon.

Fixes: https://tracker.ceph.com/issues/55265
Signed-off-by: Redouane Kachach <rkachach@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by adk3798 — APPROVED** [review]

**Comment by rkachach** [comment]
jenkins test make check

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -4682,7 +4681,7 @@ def wait_for_mon(
                              timeout=timeout)
         return ret == 0
 
-    is_available(ctx, 'mon', is_mon_available)
+    is_available(ctx, 'mon', is_mon_available, 3)
```
Comment (mgfritch) [inline src/cephadm/cephadm:4684]: > Note: as part of this fix I'm reducing the maximum tries for mon from 15 (default) to 3. So at most cephadm will wait 3 min for the mon.

do we need to reduce the retry? seems like that could be an issue on resource limited hardware (e.g. Raspberry Pi)

Comment (rkachach) [reply src/cephadm/cephadm:4684]: > > Note: as part of this fix I'm reducing the maximum tries for mon from 15 (default) to 3. So at most cephadm will wait 3 min for the mon.
> 
> do we need to reduce the retry? seems like that could be an issue on resource limited hardware (e.g. Raspberry Pi)

I expected 3 min to be enough to wait for the mon process. Could the mon process takes more in order to get up & running?

**Comment by rkachach** [comment]
jenkins test docs

**Comment by rkachach** [comment]
jenkins render docs

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-04-22_01:18:37-orch:cephadm-wip-adk-testing-2022-04-21-1828-distro-basic-smithi/

2 Failures inferring config in tests using docker, caused by another PR in the run.

File: src/cephadm/cephadm (hunk 1 above)
Comment (rkachach) [reply src/cephadm/cephadm]: @mgfritch I reverted the changes related to the retry and only kept the original fix (related to timeout).

**Comment by rkachach** [comment]
jenkins test make check

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-04-23_01:33:16-orch:cephadm-wip-adk-testing-2022-04-22-1640-distro-basic-smithi/

2 SELinux Denial Failures Tracked by https://tracker.ceph.com/issues/55443

**Review by adk3798 — APPROVED** [review]

**Pull request merged**